coconut-rs/src/traits.rs

```diff
@@ -0,0 +1,17 @@
+pub trait Bytable {
+    fn to_byte_vec(&self) -> Vec<u8>;
+
+    fn from_byte_slice(slice: &[u8]) -> Self;
```

Comment:
Seeing the example of `Attribute`, I think this should instead be changed to `try_from_byte_slice`